### PR TITLE
feat: problem type selection

### PR DESCRIPTION
**Pull request by rayzhou-bit**
https://2u-internal.atlassian.net/browse/TNL-10191

This change sets up the problem type selection page on the Problem Editor.

### CODE REVIEW

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js [hunk 1]
```diff
@@ -1,14 +1,50 @@
...
+    selected,
+    setSelected,
+  };
 };
 
-export default { state, onSelect };
+export const onSelect = ( setProblemType, selected ) => () =>
+  setProblemType({ selected });
+
+export const useArrowNav = (selected, setSelected) => {
```
Comment (rayzhou-bit) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js:21]: I messed around with tests for this but have no idea how to do it.

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx [hunk 2]
```diff
@@ -8,32 +8,34 @@ import {
...
+  setProblemType,
 }) => (
-  <div className="editor-footer" style={{ position: 'sticky', bottom: 0 }}>
+  <div className="editor-footer position-sticky" style={{ bottom: 0 }}>
     <ModalDialog.Footer className="border-top-0">
       <ActionRow>
         <ActionRow.Spacer />
         <Button
-          aria-label="TODO: CANCEL"
+          aria-label="cancel"
```
Comment (connorhaugh) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx:27]: this should be a translated string no?

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/__snapshots__/SelectTypeFooter.test.jsx.snap [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  }
+>
+  <ModalDialog.Footer
+    className="border-top-0"
+  >
+    <ActionRow>
+      <ActionRow.Spacer />
+      <Button
+        aria-label="cancel"
+        onClick={[MockFunction]}
```
Comment (connorhaugh) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/__snapshots__/SelectTypeFooter.test.jsx.snap:19]: you should give this a name

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js (hunk 1 above)
Comment (connorhaugh) [reply src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js:21]: you'd have to add a property to document, which I think you can do like this:

```
it('should test something', () => {
  const spyFunc = jest.fn();
  Object.defineProperty(global.document, 'querySelector', { value: spyFunc });
  <run some test>
  expect(spyFunc).toHaveBeenCalled()
});
```

File: src/editors/data/redux/thunkActions/problem.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ export const initializeProblem = (blockValue) => (dispatch) => {
   const { settings, ...data } = parsedProblem;
   const parsedSettings = { ...settings, ...parseSettings(_.get(blockValue, 'data.metadata', {})) };
   if (!_.isEmpty(rawOLX) && !_.isEmpty(data)) {
-    dispatch(actions.problem.load({ ...data, rawOLX, parsedSettings }));
+    dispatch(actions.problem.load({ ...data, rawOLX, settings: parsedSettings }));
```
Comment (connorhaugh) [inline src/editors/data/redux/thunkActions/problem.js:19]: good catch

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx [hunk 5]
```diff
@@ -8,32 +8,34 @@ import {
...
 import { FormattedMessage, injectIntl } from '@edx/frontend-platform/i18n';
+import messages from './messages';
 import hooks from '../hooks';
 
 import { actions } from '../../../../../data/redux';
 
 export const SelectTypeFooter = ({
-  selected,
   onCancel,
+  selected,
```
Comment (KristinAoki) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx:18]: need to add
```
// injected
intl,
```

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx [hunk 6]
```diff
@@ -47,14 +49,13 @@ SelectTypeFooter.defaultProps = {
 SelectTypeFooter.propTypes = {
   onCancel: PropTypes.func.isRequired,
   selected: PropTypes.string,
```
Comment (KristinAoki) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx:51]: add
```
intl: intlShape.isRequired,
```

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx [hunk 7]
```diff
@@ -8,32 +8,34 @@ import {
...
           variant="tertiary"
           onClick={onCancel}
         >
-          TODO: -- Cancel
+          <FormattedMessage {...messages.cancelButtonLabel} />
         </Button>
         <Button
-          aria-label="TODO: SELECT"
-          onClick={hooks.onSelect({ selected })}
+          aria-label={intl.FormattedMessage(messages.selectButtonAriaLabel)}
```
Comment (KristinAoki) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx:34]:
```suggestion
          aria-label={intl.formatMessage(messages.selectButtonAriaLabel)}
```

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx [hunk 8]
```diff
@@ -8,32 +8,34 @@ import {
...
+  setProblemType,
 }) => (
-  <div className="editor-footer" style={{ position: 'sticky', bottom: 0 }}>
+  <div className="editor-footer position-sticky" style={{ bottom: 0 }}>
     <ModalDialog.Footer className="border-top-0">
       <ActionRow>
         <ActionRow.Spacer />
         <Button
-          aria-label="TODO: CANCEL"
+          aria-label={intl.FormattedMessage(messages.cancelButtonAriaLabel)}
```
Comment (KristinAoki) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/SelectTypeWrapper/SelectTypeFooter.jsx:27]:
```suggestion
          aria-label={intl.formatMessage(messages.cancelButtonAriaLabel)}
```

**Review by KristinAoki — CHANGES_REQUESTED** [review]

File: src/editors/data/constants/problem.js [hunk 9]
```diff
@@ -20,30 +20,38 @@ export const ProblemTypes = StrictDict({
     preview: singleSelect,
     description: 'Specify one correct answer from a list of possible options',
     helpLink: 'https://edx.readthedocs.io/projects/edx-partner-course-staff/en/latest/exercises_tools/multiple_choice.html',
+    next: ProblemTypeKeys.MULTISELECT,
```
Comment (rayzhou-bit) [inline src/editors/data/constants/problem.js:23]: I added these as a node data structure for useArrowNav

File: src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.test.js [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+  });
+
+  describe('onSelect', () => {
+    test('setProblemType is called with selected', () => {
+      module.onSelect(mockSetProblemType, mockSelected)();
+      expect(mockSetProblemType).toHaveBeenCalledWith({ selected: mockSelected });
+    });
+  });
+
+  describe('useArrowNav', () => {
```
Comment (rayzhou-bit) [inline src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.test.js:50]: @connorhaugh I managed to write these tests which work. However it's definitely not cleaning up the useEffects, which could cause issues with other tests in the future.

File: src/editors/Editor.test.jsx [hunk 11]
```diff
@@ -12,7 +12,7 @@ jest.mock('./hooks', () => ({
 
 jest.mock('./containers/TextEditor', () => 'TextEditor');
 jest.mock('./containers/VideoEditor', () => 'VideoEditor');
-jest.mock('./containers/ProblemEditor/ProblemEditor', () => 'ProblemEditor');
+jest.mock('./containers/ProblemEditor', () => 'ProblemEditor');
```
Comment (farhaanbukhsh) [inline src/editors/Editor.test.jsx:15]: This fixes is alread coming in https://github.com/open-craft/frontend-lib-content-components/pull/7

**Review by KristinAoki — APPROVED** [review]

File: src/editors/Editor.test.jsx (hunk 11 above)
Comment (rayzhou-bit) [reply src/editors/Editor.test.jsx:15]: Thanks! I'm not actively seeking these out, just making changes when they're short

**Pull request merged**